test-suite/schema.json

```
@@ -0,0 +1,143 @@
+{
+  "$schema": "https://json-schema.org/draft/2020-12/schema",
+  "$defs": {
+    "scenario": {
+      "type": "object",
+      "properties": {
+        "description": {
+          "type": "string"
+        },
+        "schemas": {
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "$ref": "https://json-schema.org/draft/2020-12/schema"
+          }
+        },
+        "tests": {
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "anyOf": [
+              { "$ref": "#/$defs/hasClassAssertion" }
+            ]
+          }
+        }
+      },
+      "required": [
+        "description"
+      ],
+      "additionalProperties": false
+    },
+    "baseAssertion": {
+      "type": "object",
+      "properties": {
+        "assertion": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "assertion"
+      ]
+    },
+    "hasClassAssertion": {
+      "$ref": "#/$defs/baseAssertion",
+      "properties": {
+        "assertion": {
+          "const": "hasClass"
+        },
+        "tests": {
+          "type": "array",
+          "minItems": 1,
+          "items": {
+            "anyOf": [
+              {
+                "$ref": "#/$defs/hasNameAssertion"
+              },
+              {
+                "$ref": "#/$defs/hasPropertyAssertion"
+              }
+            ]
+          }
+        }
+      },
+      "additionalProperties": false
+    },
+    "hasNameAssertion": {
+      "$ref": "#/$defs/baseAssertion",
+      "properties": {
+        "assertion": {
+          "const": "hasName"
+        },
+        "arguments": {
+          "type": "array",
+          "prefixItems": [ {"type": "string"} ]
+        }
+      },
+      "required": [
+        "arguments"
+      ],
+      "additionalProperties": false
+    },
+    "hasPropertyAssertion": {
+      "$ref": "#/$defs/baseAssertion",
+      "properties": {
+        "assertion": {
+          "const": "hasProperty"
+        },
+        "tests": {
+          "type": "array",
+          "prefixItems": [
+            {
+              "$ref": "#/$defs/hasNameAssertion"
+            },
+            {
+              "$ref": "#/$defs/hasTypeAssertion"
+            }
+          ]
```

Did you really mean to use `prefixItems` here?

```suggestion
          "items": {
            "anyOf": [
              {
                "$ref": "#/$defs/hasNameAssertion"
              },
              {
                "$ref": "#/$defs/hasTypeAssertion"
              }
            ]
          }
```

---

That's a good point, changing 🙂